### DISCUSSION TITLE
How do i send an embed as a slash command?

### DISCUSSION THREAD

**Discussion by RealMrCactus** [serenity-rs/serenity#2936] [Q&A] [answered]
this is all i could figure out.

```rs
use serenity::all::Context;
use serenity::builder::CreateCommand;
use serenity::builder::{CreateEmbed, CreateEmbedFooter, CreateMessage};
use serenity::model::Timestamp;

pub fn run(ctx: Context) -> Result<(), serenity::Error> {
    let footer = CreateEmbedFooter::new("This is a footer");
    let embed = CreateEmbed::new()
        .title("This is a title")
        .description("This is a description")
        .fields(vec![
            ("This is the first field", "This is a field body", true),
            ("This is the second field", "Both fields are inline", true),
        ])
        .field("This is the third field", "This is not an inline field", false)
        .footer(footer)
        // Add a timestamp for the current time
        // This also accepts a rfc3339 Timestamp
        .timestamp(Timestamp::now());
    let builder = CreateMessage::new()
                .content("Hello, World!")
                .embed(embed);
    Ok(())
}

pub fn register() -> CreateCommand {
    CreateCommand::new("embed").description("Embeds a message")
}
```

**Comment by jamesbt365** [comment, marked as answer]
It is much easier for you to use [poise](https://github.com/serenity-rs/poise), but if you really don't want to, and haven't already, check out the example for doing slash commands in [serenity](https://github.com/serenity-rs/serenity/tree/current/examples/e14_slash_commands).

Doing it directly in serenity is not pretty by any means, you have to:
- Build the structure and register your commands manually
- Handle the `InteractionCreate` event, match the enum for `CommandInteraction`
- Parse the interaction to get the arguments out manually
- Then send a response either through the `create_response` method or the `create_followup` method, depending on if you have already sent a response or not.

It certainly isn't pretty, but if you don't want to use poise that is how you have to do it. See the example for a full rundown.
